soh/soh/Enhancements/randomizer/3drando/random.cpp

```diff
@@ -1,36 +1,57 @@
 #include "random.hpp"
 
+#include <numeric>
 #include <random>
 #include <boost/random/mersenne_twister.hpp>
 #include <boost/random/uniform_int_distribution.hpp>
 #include <boost/random/uniform_real_distribution.hpp>
 
 static bool init = false;
-static boost::random::mt19937 generator;
+static uint64_t state = 0;
+const uint64_t multiplier = 6364136223846793005ULL;
+const uint64_t increment = 11634580027462260723ULL;
 
 //Initialize with seed specified
-void Random_Init(uint32_t seed) {
+void Random_Init(uint64_t seed) {
     init = true;
-    generator = boost::random::mt19937{seed};
+    state = seed;
 }
 
-//Returns a random integer in range [min, max-1]
-uint32_t Random(int min, int max) {
+uint32_t next32() {
     if (!init) {
         //No seed given, get a random number from device to seed
 #if !defined(__SWITCH__) && !defined(__WIIU__)
-        const auto seed = static_cast<uint32_t>(std::random_device{}());
+        uint64_t seed = static_cast<uint64_t>(std::random_device{}());
 #else
-        uint32_t seed = static_cast<uint32_t>(std::hash<std::string>{}(std::to_string(rand())));
+        uint64_t seed = static_cast<uint64_t>(std::hash<std::string>{}(std::to_string(rand())));
```

Comment:
```suggestion
        uint64_t seed = static_cast<uint64_t>(rand());
```
This string hash dance is ultimately pointless. Some checking of RAND_MAX could find it better to do `(uint64_t)rand() | (uint64_t)rand() << 32`

Reply:
i know this PR is closed but i do want to note something:

that line is behind a switch/wii u ifdef

as for how it got there, we originally had

https://github.com/HarbourMasters/Shipwright/blob/7fb8902e7fce15d86c29d11790bada85e9fd352a/soh/soh/Enhancements/randomizer/3drando/random.cpp#L15-L23

which was added in https://github.com/HarbourMasters/Shipwright/pull/416

then it was updated to

https://github.com/HarbourMasters/Shipwright/blob/7df9641297e8ac5965ed9bfd62dc98fae38d6a09/soh/soh/Enhancements/randomizer/3drando/random.cpp#L18-L30

in https://github.com/HarbourMasters/Shipwright/pull/1608

not sure *why* the "dance" was used as a fix for wii u but that's the origin story at least